Mathlib/Topology/Constructible.lean

```diff
@@ -0,0 +1,300 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Order.BooleanSubalgebra
+import Mathlib.Topology.Spectral.Hom
+
+/-!
+# Constructible sets
+
+This file defines constructible sets, which are morally sets in a topological space which we can
+make out of unions and intersections of open and closed sets.
```

Comment:
```suggestion
make out of finite unions and finite intersections of open and closed sets.
```
important distinction!

Reply:
The "morally" is the fact that in full generality open and closed sets should be replaced by open retrocompact and complement of open retrocompact sets, which is a mouthful for what's ultimately a technicality

Reply:
(It's not a technicality, it's fundamental: in the spectrum of a ring $R$, the complement of a retrocompact open set is of the form $V(f_1,\dots,f_r)$ for finitely many elements $f_i$ of $R$. Constructible sets would be uninteresting. The parallel with first order model theory is perfect: formulas are finite. Infinite logic exists, but it's a completely different story.)

Reply:
Any plans to connect to [IsLocallyClosed](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Defs/Basic.html#IsLocallyClosed)? [Wikipedia](https://en.wikipedia.org/wiki/Constructible_set_(topology)#Definitions) claims that in a locally noetherian space (Mathlib only has noetherian spaces), all subsets are retrocompact, and the constructible sets coincide with finite unions of locally closed sets.

Reply:
No, no plan. Feel free to once it's merged!